packages/backend/src/core/transaction-count/ZksyncTransactionUpdater.ts

```diff
@@ -0,0 +1,78 @@
+import { Logger, TaskQueue, UniqueTaskQueue } from '@l2beat/common'
+
+import {
+  ZksyncTransactionRecord,
+  ZksyncTransactionRepository,
+} from '../../peripherals/database/ZksyncTransactionRepository'
+import { ZksyncClient } from '../../peripherals/zksync'
+import { Clock } from '../Clock'
+
+export class ZksyncTransactionUpdater {
+  private updateQueue = new TaskQueue<void>(() => this.update(), this.logger)
+  private blockQueue = new UniqueTaskQueue(
+    this.updateBlock.bind(this),
+    this.logger,
+    {
+      workers: 100,
+    },
+  )
+
+  constructor(
+    private zksyncClient: ZksyncClient,
+    private zksyncTransactionRepository: ZksyncTransactionRepository,
+    private clock: Clock,
+    private logger: Logger,
+  ) {
+    this.logger = logger.for(this)
+  }
+
+  start() {
+    this.logger.info('Started')
+    this.updateQueue.addIfEmpty()
+    return this.clock.onNewHour(() => {
+      this.updateQueue.addIfEmpty()
+    })
+  }
+
+  async updateBlock(number: number) {
+    const transactions = await this.zksyncClient.getTransactionsInBlock(number)
+
+    const records: ZksyncTransactionRecord[] = transactions.map(
+      (transaction) => ({
+        blockIndex: transaction.blockIndex,
+        blockNumber: number,
+        timestamp: transaction.createdAt,
+      }),
+    )
```

Comment:
If a block had 0 transactions it would be missing from the database and therefore always re-downloaded. Are there any such blocks?

Reply:
I've found not such blocks up to 100k (now at 107k)